src/applications/vaos/services/appointment/index.js

```diff
@@ -119,47 +119,66 @@ function hasPartialResults(response) {
 export async function getBookedAppointments({
   startDate,
   endDate,
-  useV2 = false,
+  useV2VA = false,
+  useV2CC = false,
 }) {
   try {
-    if (useV2) {
-      const appointments = await getAppointments(startDate, endDate, [
-        'booked',
-        'cancelled',
-      ]);
-
-      const appointmentsWithoutRequests = appointments.filter(
-        appt => !appt.requestedPeriods,
+    let appointments = [];
+    if (!useV2VA) {
+      const confirmedVAAppointments = await getConfirmedAppointments(
+        'va',
+        moment(startDate).toISOString(),
+        moment(endDate).toISOString(),
       );
 
-      return transformVAOSAppointments(appointmentsWithoutRequests);
+      // We might get partial results back from MAS, so throw an error if we do
+      if (hasPartialResults(confirmedVAAppointments)) {
+        throw mapToFHIRErrors(
+          appointments[0].errors,
+          'MAS returned partial results',
+        );
+      }
+
+      appointments = [
```

Comment:
If you want to avoid reassigning each time, you could do `appointments.push(...transformConfirmedAppointments(confirmedVAAppointments.data))`